### ISSUE TITLE
comma in template string right after ${} causes comma-spacing

### ISSUE THREAD

**Issue by dimsmol** [babel/babel-eslint#31]
minimal example:
```
let a = 1;
console.log(`${a}, b`); // will report comma-spacing
```
comma escaping can be used as a workaround:
```
let a = 1;
console.log(`${a}\, b`); // no error
```

**Comment by gaearon** [comment]
Yep, same problem in JSX text content.

**Comment by douglasduteil** [comment]
Same

**Comment by mkawalec** [comment]
We have the same problem.

**Comment by sebmck** [comment]
Can you please verify that this issue still exists on 3.0.1?

**Comment by wavded** [comment]
Using 3.0.1 I still encounter the issue with commas either in template strings or in JSX.

**Comment by abelnation** [comment]
I still see it as well.

src line:
`console.log(`${ios}, ${android}, ${phone}, ${tablet}`)`

error:
`  267:27  error  A space is required after ','  comma-spacing`

**Comment by zdychacek** [comment]
Still not working, the same error:

```log.info(`Failed to enable text ad ${advertId}, trying to enable banner.`, textAdException);```

**Comment by atombender** [comment]
Same here, still getting it with 3.0.1.

**Comment by hzoo** [comment]
Looking into this... The tokens in espree/esprima aren't the same?

```js
// espree (has no comma node) and thus no issue
[ { type: 'Template', value: '`${' },
  { type: 'Identifier', value: 'a' },
  { type: 'Template', value: '}, b`' },
  { type: 'Punctuator', value: ';' } ]
// babel/acorn
[ { type: 'Punctuator', value: '`' },
  { type: 'Punctuator', value: 'template' },
  { type: 'Punctuator', value: '${' },
  { type: 'Identifier', value: 'a' },
  { type: 'Punctuator', value: '}' },
  { type: 'Punctuator', value: ', b' },
  { type: 'Punctuator', value: '`' },
  { type: 'Punctuator', value: ';' } ]
```
So maybe we need to combine tokens to make a `Template` type? @sebmck

**Comment by hzoo** [comment]
Ok `comma-spacing` in jsx text should be fixed in 3.1.5.

**Comment by MoOx** [comment]
I just tried 3.1.5 and still get issue on this snippet


```js
                  (
                    `upload/`,
                    `upload/` +
                      `w_${ Math.round(item.crop.width / item.crop.ratio) },` +
                      `h_${ Math.round(item.crop.height / item.crop.ratio) }/` +
                      `x_${ Math.round(item.crop.x / item.crop.ratio) },` +
                      `y_${ Math.round(item.crop.y / item.crop.ratio) },` +
                      `w_${ Math.round(item.crop.cropWidth) },` +
                      `h_${ Math.round(item.crop.cropHeight) },` +
                      `c_crop/`
                  )
```

**Comment by hzoo** [comment]
Ok I got it down to

```js
a(`a`,
  `a`)
```

So does it error on the eslint parser? I think it's erroring because it doesn't think there is a newline.

**Comment by MoOx** [comment]
> on the eslint parser 

I use babel-eslint with probably too many things that might break eslint default parser

No eslint exceptions if it's what you are asking.

I just get "normal" `comma-spacing` reported.

**Comment by hzoo** [comment]
Ok I copied the snippet into a new file and don't get an issue?

```js
{
  "env": {
    "es6": true,
    "node": true
  },
  "parser": "babel-eslint",
  "rules": {
    "comma-spacing": 1
  }
}
```

**Comment by MoOx** [comment]
hum ok I will investigate that later. Sorry if it's a mistake from my side.

**Comment by XeCycle** [comment]
Bump, this is still complaining:
```js
module.exports = {
  render() {
    var {name} = this.props;
    return Math.max(null, `Name: ${name}, Name: ${name}`);
  }
};
```
Trimmed down from a React component code to this.  Warning would go away if changed `var {name}` to `var name`.

**Comment by hzoo** [comment]
Ah sorry can reproduce - will push a change soon.

**Comment by KylePDavis** [comment]
Thanks, @hzoo. That fixes this for me:

**Comment by hzoo** [comment]
@KylePDavis Awesome!
@XeCycle Sorry, I was going to wait for another fix but it's taking a while, so that should be fixed in 3.1.8!
